src/components/Hub/HubImagesList.tsx

```diff
@@ -12,28 +11,118 @@ import ImageCard from "./ImageCard"
 import HubFilters, { getSelectedFilters } from "./HubFilters"
 import SpinningLoader from "../Common/SpinningLoader"
 import { ExpandingSearchbar } from "../Common/ExpandingSearchbar"
-import { FilterCategory,  FilterParams } from "../../redux/hub/hub.types"
+import {
+  FilterCategory,
+  FilterParams,
+  HubImage,
+} from "../../redux/hub/hub.types"
 import styled from "@emotion/styled"
+import { Grid } from "@material-ui/core"
+import HubSortDropdown from "./HubSortDropdown"
+import { sortOptions } from "../../redux/hub/hub.constants"
 
-const SearchContainer = styled(Row)`
-  flex-direction: row-reverse;
-  padding: 1rem;
-`
+const sortHubImages = (images: HubImage[], selectedSortOption: string) => {
+  switch (selectedSortOption) {
+    case "name increasing":
+      return images.sort((imageA, imageB) =>
+        imageA.name < imageB.name ? -1 : imageA.name > imageB.name ? 1 : 0
+      )
+    case "name decreasing":
+      return images.sort((imageA, imageB) =>
+        imageA.name > imageB.name ? -1 : imageA.name < imageB.name ? 1 : 0
+      )
+    case "author increasing":
+      return images.sort((imageA, imageB) =>
+        imageA.author < imageB.author
+          ? -1
+          : imageA.author > imageB.author
+          ? 1
+          : 0
+      )
+    case "author decreasing":
+      return images.sort((imageA, imageB) =>
+        imageA.author > imageB.author
+          ? -1
+          : imageA.author < imageB.author
+          ? 1
+          : 0
+      )
+  }
+}
+
+const HubLibraryContainer: React.FC = ({ children }) => {
```

Comment:
I'm a bit confused why these have to be encapsulated in React function components

Reply:
I did this in order to set the container property. Can you think of a more slim way to do that?